src/v/security/gssapi_principal_mapper.cc

```diff
@@ -33,87 +34,10 @@ static constexpr std::string_view gssapi_name_pattern
   = R"(([^/@]*)(/([^/@]*))?@([^/@]*))";
 static constexpr std::string_view non_simple_pattern = R"([/@])";
 static constexpr std::string_view parameter_pattern = R"(([^$]*)(\$(\d*))?)";
-static constexpr std::string_view rule_pattern
-  = R"(((DEFAULT)|((RULE:\[(\d*):([^\]]*)](\(([^)]*)\))?(s\/([^\/]*)\/([^\/]*)\/(g)?)?\/?(L|U)?))))";
 
 namespace detail {
 std::vector<gssapi_rule>
-parse_rules(const std::vector<ss::sstring>& unparsed_rules) {
-    static thread_local const re2::RE2 rule_parser(
-      rule_pattern, re2::RE2::Quiet);
-
-    vassert(
-      rule_parser.ok(),
-      "Failed to build rule pattern regex: {}",
-      rule_parser.error());
-
-    if (unparsed_rules.empty()) {
-        return {gssapi_rule()};
-    }
-
-    std::vector<gssapi_rule> rv;
-    re2::StringPiece default_;
-    re2::StringPiece num_components_str;
-    re2::StringPiece format;
-    re2::StringPiece match_regex;
-    re2::StringPiece from_pattern;
-    re2::StringPiece to_pattern;
-    re2::StringPiece repeat;
-    re2::StringPiece upper_lower;
-    for (const auto& rule : unparsed_rules) {
-        const re2::StringPiece rule_piece(rule.data(), rule.size());
-        if (!re2::RE2::FullMatch(
-              rule_piece,
-              rule_parser,
-              nullptr,
-              &default_,
-              nullptr,
-              nullptr,
-              &num_components_str,
-              &format,
-              nullptr,
-              &match_regex,
-              nullptr,
-              &from_pattern,
-              &to_pattern,
-              &repeat,
-              &upper_lower)) {
-            throw std::runtime_error("GSSAPI: Invalid rule: " + rule);
-        }
-        if (!default_.empty()) {
-            rv.emplace_back();
-        } else {
-            int num_components = std::numeric_limits<int>::max();
-            auto conv_rc = std::from_chars(
-              num_components_str.begin(),
-              num_components_str.end(),
-              num_components);
-            if (conv_rc.ec != std::errc()) {
-                throw std::runtime_error(
-                  "Invalid rule - Invalid value for number of components: "
-                  + num_components_str.as_string());
-            }
-            gssapi_rule::case_change_operation case_change
-              = gssapi_rule::case_change_operation::noop;
-
-            if (upper_lower == "L") {
-                case_change = gssapi_rule::case_change_operation::make_lower;
-            } else if (upper_lower == "U") {
-                case_change = gssapi_rule::case_change_operation::make_upper;
-            }
-            rv.emplace_back(
-              num_components,
-              spv(format),
-              spv(match_regex),
-              spv(from_pattern),
-              spv(to_pattern),
-              gssapi_rule::repeat{repeat == "g"},
-              case_change);
-        }
-    }
-
-    return rv;
-}
+parse_rules(const std::vector<ss::sstring>& unparsed_rules);
```

Comment:
comment: It's kind of hard to which `parse_rules` is being used.  Yes I can see that they are in different namespaces, but when the function call (or declaration) is heavily nested it's hard to see.
suggestion: Maybe the `parse_rules` for gssapi can be moved into its own namespace (`swecurity::gssapi::detail`?), like how the `parse_rules` for tls is in `security::tls::detail`.